http-client-core/src/main/java/io/micronaut/http/client/bind/binders/QueryValueClientArgumentBinder.java

```diff
@@ -0,0 +1,185 @@
+                .orElse(QueryValue.Format.COMMA_DELIMITED);
+        MutableHttpParameters parameters = request.getParameters();
+
+        if (format == QueryValue.Format.DEEP_OBJECT) {
+            addDeepObjectParameters(context, value, key, parameters);
+        } else if (format == QueryValue.Format.MULTI) {
+            addMultiParameters(context, value, key, parameters);
+        } else {
+            String delimiter = "";
+            switch(format) {
+                case SPACE_DELIMITED:
+                    delimiter = SPACE_DELIMITER;
+                    break;
+                case PIPE_DELIMITED:
+                    delimiter = PIPE_DELIMITER;
+                    break;
+                case COMMA_DELIMITED:
+                    delimiter = COMMA_DELIMITER;
+                    break;
+            }
+            createSeparatedQueryValue(context, value, delimiter)
+                    .ifPresent(v -> parameters.add(key, v));
+        }
+    }
+
+    private void addMultiParameters(
+            ArgumentConversionContext<Object> context, Object value, String key, MutableHttpParameters parameters
+    ) {
+        if (value instanceof Iterable) {
+            // noinspection unechecked
+            Iterable<Object> iterable = (Iterable<Object>) value;
+
+            for (Object item : iterable) {
+                valueToString(context, item).ifPresent(v -> parameters.add(key, v));
+            }
+        } else {
+            valueToString(context, value).ifPresent(v -> parameters.add(key, v));
+        }
+    }
+
+    private void addDeepObjectParameters(
+            ArgumentConversionContext<Object> context, Object value, String key, MutableHttpParameters parameters
+    ) {
+        if (value instanceof Iterable) {
+            StringBuilder builder = new StringBuilder(key);
+
+            // noinspection unechecked
+            Iterable<Object> iterable = (Iterable<Object>) value;
+
+            int i = 0;
+            for (Object item: iterable) {
+                if (item == null) {
+                    continue;
+                }
+                String index = String.valueOf(i);
+
+                builder.append('[');
+                builder.append(index);
+                builder.append(']');
+
+                valueToString(context, item).ifPresent(v -> parameters.add(builder.toString(), v));
+                builder.delete(builder.length() - index.length() - 2, builder.length());
+                i++;
+            }
+        } else if (value != null) {
+            StringBuilder builder = new StringBuilder(key);
+            // noinspection unechecked
+            BeanWrapper wrapper = BeanWrapper.getWrapper(value);
+            // noinspection unchecked
+            Collection<BeanProperty<Object,Object>> properties = wrapper.getBeanProperties();
+            for (BeanProperty<Object, Object> property: properties) {
+                Object item = property.get(value);
```

Comment:
Need to check if `property.isWriteOnly()` and skip it if it is otherwise this will throw an exception